### DISCUSSION TITLE
Is it possible to use a checkpoint (EN) to train another language?

### DISCUSSION THREAD

**Discussion by egorsmkv** [p0p4k/vits2_pytorch#51] [Q&A] [answered]
I have a language with a different symbol table. So I need to train a checkpoint from the ground or I can use already trained one?

**Comment by p0p4k** [comment, marked as answer]
Yes;
1) load the model with random weights and new vocab -- `A`
2) load the pretrained model with older vocab (and therefore different `TextEncoder` embedding layer). -- `B`
3) copy the weights from `B` to `A` in all the layers that are same size and keep others random.

**Reply by p0p4k** [reply]
Right now since the original vits repo did not have this kind of code, I did not add it in this repo. I might add it later on.

**Reply by egorsmkv** [reply]
If you could add this code it would be awesome!

**Reply by egorsmkv** [reply]
@p0p4k hi! have you been working on it?

**Reply by p0p4k** [reply]
https://github.com/anonymous-pits/pits/blob/main/utils.py
You can refer that load checkpoint diff size and load checkpoint functions for reference.

**Reply by lumpidu** [reply]
@egorsmkv : so how was the outcome ? Have you tried it ?

**Reply by egorsmkv** [reply]
> @egorsmkv : so how was the outcome ? Have you tried it ?

No, I haven't tried it yet but plan to
